### PR TITLE
Adding GitLab Docs extension

### PR DESCRIPTION
**Pull request by saschaeggi**
## Description

As an employee of GitLab I'd like to add this new GitLab Docs extension which adds the ability to search GitLab for the following:

- GitLab Docs
- GitLab Handbook
- Pajamas Design System (GitLab's Design System)

Original repo: https://gitlab.com/seggenberger/raycast-gitlab-docs

## Screencast

https://user-images.githubusercontent.com/5558166/176434318-9470e223-b221-436c-981a-9337d53429b0.mp4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/gitlab-docs/src/docs.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+      icon="../assets/docs-icon.png"
+      title={searchResult.name}
+      subtitle={searchResult.description}
+      accessoryTitle={searchResult.category}
+      actions={
+        <ActionPanel>
+          <ActionPanel.Section>
+            <Action.OpenInBrowser title="Show Details" url={searchResult.url} />
+          </ActionPanel.Section>
+          <ActionPanel.Section>
```
Comment (pernielsentikaer) [inline extensions/gitlab-docs/src/docs.tsx:34]:
```suggestion
```

File: extensions/gitlab-docs/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    <List.Item
+      icon="../assets/handbook-icon.png"
+      title={searchResult.name}
+      accessoryTitle={searchResult.category}
+      actions={
+        <ActionPanel>
+          <ActionPanel.Section>
+            <Action.OpenInBrowser title="Show Details" url={searchResult.url} />
+          </ActionPanel.Section>
+          <ActionPanel.Section>
```
Comment (pernielsentikaer) [inline extensions/gitlab-docs/src/index.tsx:37]:
```suggestion
```

File: extensions/gitlab-docs/src/pajamas.tsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    <List.Item
+      icon="../assets/pajamas-icon.png"
+      title={searchResult.name}
+      accessoryTitle={searchResult.category}
+      actions={
+        <ActionPanel>
+          <ActionPanel.Section>
+            <Action.OpenInBrowser title="Show Details" url={searchResult.url} />
+          </ActionPanel.Section>
+          <ActionPanel.Section>
```
Comment (pernielsentikaer) [inline extensions/gitlab-docs/src/pajamas.tsx:39]:
```suggestion
```

File: extensions/gitlab-docs/src/pajamas.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+function SearchListItem({ searchResult }: { searchResult: SearchResult }) {
+  return (
+    <List.Item
+      icon="../assets/pajamas-icon.png"
+      title={searchResult.name}
+      accessoryTitle={searchResult.category}
+      actions={
+        <ActionPanel>
+          <ActionPanel.Section>
+            <Action.OpenInBrowser title="Show Details" url={searchResult.url} />
```
Comment (pernielsentikaer) [inline extensions/gitlab-docs/src/pajamas.tsx:37]: This title is usually used when opening the Detail Panel; maybe use the default title?

File: extensions/gitlab-docs/src/index.tsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+function SearchListItem({ searchResult }: { searchResult: SearchResult }) {
+  return (
+    <List.Item
+      icon="../assets/handbook-icon.png"
+      title={searchResult.name}
+      accessoryTitle={searchResult.category}
+      actions={
+        <ActionPanel>
+          <ActionPanel.Section>
+            <Action.OpenInBrowser title="Show Details" url={searchResult.url} />
```
Comment (pernielsentikaer) [inline extensions/gitlab-docs/src/index.tsx:35]: This title is usually used when opening the Detail Panel; maybe use the default title?

File: extensions/gitlab-docs/src/docs.tsx [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+  return (
+    <List.Item
+      icon="../assets/docs-icon.png"
+      title={searchResult.name}
+      subtitle={searchResult.description}
+      accessoryTitle={searchResult.category}
+      actions={
+        <ActionPanel>
+          <ActionPanel.Section>
+            <Action.OpenInBrowser title="Show Details" url={searchResult.url} />
```
Comment (pernielsentikaer) [inline extensions/gitlab-docs/src/docs.tsx:32]: This title is usually used when opening the Detail Panel; maybe use the default title?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @saschaeggi 👋 

Thanks for your contribution 🙂

I have some feedback for you to look into:

- Is it possible for Docs and Handbook to show some initial data like Pajamas does?

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by saschaeggi** [comment]
@pernielsentikaer I've updated Docs & Handbook to save the previous search. So it always shows the last search results instead of the empty screen. Also if nothing has been searched for it shows placeholder results.

For the Pajamas search: I grouped them by category, as we have them anyway in the Design System.

Can be reviewed again!

**Comment by saschaeggi** [comment]
@pernielsentikaer any update on this?

File: extensions/gitlab-docs/package.json [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  "name": "gitlab-docs",
+  "title": "GitLab Docs",
+  "description": "Raycast extension to search GitLab Docs, Handbook & Pajamas Design System.",
+  "icon": "gitlab-icon.png",
+  "author": "saschaeggi",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "GitLab Handbook",
```
Comment (pernielsentikaer) [inline extensions/gitlab-docs/package.json:12]:
```suggestion
      "title": "Search Handbook",
      "subtitle": "GitLab",
```

File: extensions/gitlab-docs/package.json [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+      "description": "Search GitLab Handbook.",
+      "mode": "view",
+      "icon": "handbook-icon.png"
+    },
+    {
+      "name": "docs",
+      "title": "GitLab Docs",
```
Comment (pernielsentikaer) [inline extensions/gitlab-docs/package.json:19]:
```suggestion
      "title": "Search Documentation",
      "subtitle": "GitLab",
```

File: extensions/gitlab-docs/package.json [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+      "description": "Search GitLab Docs.",
+      "mode": "view",
+      "icon": "docs-icon.png"
+    },
+    {
+      "name": "pajamas",
+      "title": "Pajamas Design System",
```
Comment (pernielsentikaer) [inline extensions/gitlab-docs/package.json:26]:
```suggestion
      "title": "Search Design System",
      "subtitle": "Pajamas",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

I gave it a new spin and have a few comments for you 🙂

- Could you add [metadata images](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots)? With our new tool, it should be easy doing so 🙂

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by saschaeggi** [comment]
@pernielsentikaer can be reviewed again 🙂

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**